homework-submission/React_HW2/todoApp2.js

```diff
@@ -0,0 +1,149 @@
+'use strict';
+
+// ContentDynamicTodo component is for map function to use
+const ContentDynamicTodo = ({ description, deadline, className, deleteItem }) => {
+  return (
+    <tbody>
+      <tr className={className} onClick={deleteItem}>
+        <td class="evenBackgroundColor">
+          <td class="content">{description}</td>
+          <td class="content">{deadline}</td>
+          <td class="content">{className}</td>
+        </td>
+      </tr>
+    </tbody>
+  );
+};
+
+// DynamicList where map function is
+const DynamicList = ({ list, deleteItem }) => {
+  const runTodoList = list.map(todo => (
+    <ContentDynamicTodo
+      key={todo.id}
+      description={todo.description}
+      deadline={todo.deadline}
+      className={todo.done ? '√' : 'x'}
+      deleteItem={() => {
+        deleteItem(todo.id);
+      }}
+    />
+  ));
+  return <tbody>{runTodoList}</tbody>;
+};
+
+// Here is our main Application component as a class
+class AppTodo2 extends React.Component {
+  //TodoList which contains new item as state
+  constructor(props) {
+    super(props);
+    this.state = {
+      todoList: [
+        {
+          id: 1,
+          description: 'Get out of bed',
+          deadline: '2017-09-11',
+          done: true,
+        },
+        {
+          id: 2,
+          description: 'Brush teeth',
+          deadline: '2017-09-10',
+          done: false,
+        },
+        {
+          id: 3,
+          description: 'Eat breakfast',
+          deadline: '2017-09-09',
+          done: false,
+        },
+      ],
+      newTodo: {
+        id: '',
+        description: 'Description',
+        deadline: 'No Deadline Entry',
+        done: false,
+      },
+    };
+  }
+
+  // Adding a description for new todo.
+  NewTodoDescription = () => {
+    return (
+      <p>
+        <label htmlFor="">Description </label>
+        <div id="containerDescriptionInput">
+          <input
+            type="text"
+            onChange={event => {
```

Comment:
it's fine to keep a complete function here for _demonstrating_ purposes, but not in a production!
this will be a confusing part for anyone reading the code.

Usually assigning a value will be once and forever, especially when we are dealing with ids, making them very unique to the item and not any other, you are doing that two times first on line 68 and the second on line 86.

and you are doing that on any change so if no changes happens and you wished to have the same exact content of the previous item here is how it goes:


![ezgif com-video-to-gif-1](https://user-images.githubusercontent.com/25272822/60750262-4acf5500-9fa6-11e9-8256-bf4819ddcd19.gif)


And note that the console will complain about `Warning: Encountered two children with the same key`

So I would leave this event arrow function here but the id submission in a click event attached to the add button instead!
This will help keeping it:
 1. added once 
 2. submitting as many as wanted on every click since it's a click event 🙂

Reply:
Since I put Id two times  id: Math.random() that is the main reason.
When I combine this two the problem will be solved.

Reply:
Looking forward to see the solution :)